includes/fbproduct.php

```diff
@@ -447,6 +464,62 @@ public function get_fb_description() {
 		return apply_filters( 'facebook_for_woocommerce_fb_product_description', $description, $this->id );
 	}
 
+	public function get_rich_text_description() {
+		$rich_text_description = '';
+
+		if ( $this->fb_description ) {
+			$rich_text_description = $this->fb_description;
+		}
+
+		if ( $this->rich_text_description ) {
+			$rich_text_description = $this->rich_text_description;
+		}
+		
+		
+
+		if ( empty( $rich_text_description ) ) {
+			// Try to get description from post meta if fb description has been set
+			$rich_text_description = get_post_meta(
+				$this->id,
+				self::FB_PRODUCT_DESCRIPTION,
+				true
+			);
+			if ($rich_text_description){
+                self::$rich_text_description_source = WC_Facebookcommerce_Utils::FB_DESCRIPTION;
+            }
+		}
+
+		// For variable products, we want to use the rich text description of the variant.
+		// If that's not available, fall back to the main (parent) product's rich text description.
+		if (empty($rich_text_description) && WC_Facebookcommerce_Utils::is_variation_type($this->woo_product->get_type())) {
+			$rich_text_description = WC_Facebookcommerce_Utils::clean_string($this->woo_product->get_rich_text_description());
+
+			// If the variant's rich text description is still empty, use the main product's rich text description as a fallback.
+			if (empty($rich_text_description) && $this->main_description) {
+				$rich_text_description = $this->rich_text_description;
+			}
+		}
+
+		// If no description is found from meta or variation, get from post
+		if ( empty( $rich_text_description ) ) {
```

Comment:
I can see there are a lot of conditions here, ideally each of these conditions should be covered by tests and clearly highlighted in dogfooding document and should be manually tested by QA.
I don't see variants and post_data conditions in the unit tests below. Please update the tests.